core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -269,7 +269,9 @@ private void doExecute0(ClientRequestContext ctx, HttpRequestDuplicator rootReqD
 
         derivedCtx.log().whenAvailable(RequestLogProperty.RESPONSE_HEADERS).thenAccept(log -> {
             try {
-                if (needsContentInStrategy) {
+                final Throwable responseCause =
+                        log.isAvailable(RequestLogProperty.RESPONSE_CAUSE) ? log.responseCause() : null;
+                if (needsContentInStrategy && responseCause == null) {
```

Comment:
What if an exception is raised when subscribing to `HttpResponse`?
Shouldn't we use always use duplicator if `needsContentInStrategy` is true?

Reply:
As mentioned in the PR description, HttpResponse was aborted before subscription, then the response could not be duplicated. So HttpResponse is duplicated if `responseCause` is not null.